src/zenml/config/compiler.py

```diff
@@ -643,12 +646,50 @@ def _compute_pipeline_spec(
                 "https://docs.zenml.io/user-guides/starter-guide"
             )
 
-        additional_spec_args: Dict[str, Any] = {
-            "source": pipeline.resolve(),
-            "parameters": pipeline._parameters,
-        }
+        output_specs = [
+            OutputSpec(
+                step_name=output_artifact.invocation_id,
+                output_name=output_artifact.output_name,
+            )
+            for output_artifact in pipeline._output_artifacts
+        ]
+        output_schema = compute_pipeline_output_schema(pipeline=pipeline)
+
+        return PipelineSpec(
+            steps=step_specs,
+            outputs=output_specs,
+            output_schema=output_schema,
+            source=pipeline.resolve(),
+            parameters=pipeline._parameters,
+        )
+
 
-        return PipelineSpec(steps=step_specs, **additional_spec_args)
+def compute_pipeline_output_schema(
+    pipeline: "Pipeline",
+) -> Optional[Dict[str, Any]]:
+    """Computes the pipeline output schema.
+
+    Args:
+        pipeline: The pipeline for which to compute the output schema.
+
+    Returns:
+        The pipeline output schema.
+    """
+    if not pipeline._output_artifacts:
+        return None
+
+    output_model_class: Type[BaseModel] = create_model(
+        f"{pipeline.name}_output",
+        __config__=ConfigDict(arbitrary_types_allowed=True),
+        **{
+            output_artifact.output_name: (
+                output_artifact.annotation.resolved_annotation,
+                ...,
+            )
+            for output_artifact in pipeline._output_artifacts
+        },
+    )
+    return output_model_class.model_json_schema(mode="serialization")
```

Comment:
@stefannica @safoinme And a second question here: How should we treat this if the output's contain a type that is not JSON serializable? Don't store any output schema at all I assume?